auth/providers/azure/pop_tokenverifier.go

```diff
@@ -136,6 +143,28 @@ func (p *PoPTokenVerifier) ValidatePopToken(token string) (string, error) {
 		return "", errors.Errorf("Invalid token. 'u' claim is missing")
 	}
 
+	// Verify host 'nonce' claim
+	if nonce, ok := claims["nonce"]; ok {
+		if _, ok := nonce.(string); !ok {
+			return "", errors.Errorf("Invalid token. 'nonce' claim should be of string")
+		}
+	} else {
+		return "", errors.Errorf("Invalid token. 'nonce' claim is missing")
+	}
+	// Making sure nonce is not reused
+	if _, ok := nonceMap[claims["nonce"].(string)]; ok {
+		return "", errors.Errorf("Invalid token. 'nonce' claim is reused")
+	}
+	nonceMap[claims["nonce"].(string)] = time.Now()
+	nonceMapCounter++
+	klog.V(6).Infof("nonce claim added to the cache. Cache size is: %d", nonceMapCounter)
+	// Cleaning cached nonce token after PoPTokenValidityDuration minutes + 1 minute
+	go func() {
+		time.Sleep((p.PoPTokenValidityDuration + 1) * time.Minute)
+		delete(nonceMap, claims["nonce"].(string))
+		nonceMapCounter--
```

Comment:
Updating `nonceMapCounter` var should be made thread safe to prevent competing updates from multiple go routines.